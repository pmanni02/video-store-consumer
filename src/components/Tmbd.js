import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../App.css';
import axios from 'axios';
import SearchTmbdForm from './SearchTmbdForm.js'
import ShowMovieResults from './ShowMovieResults.js'

class Tmbd extends Component {
  constructor(){
    super();
    this.state = {
      movies: []
    }
  }

  getMovies = (search) => {
    const url = 'http://localhost:3000//movies?query=';
    axios.get(`${url}${search}`)
      .then((response) => {
        this.setState({
          movies: response.data
        });
        console.log(this.state.movies);
      })
      .catch((error) => {
        console.log(error);
      })
  }

  statusUpdate = (status) => {
    this.props.statusCallback(status);
  }

  render(){
    return(
      <div>
        <div>
          <SearchTmbdForm
            searchCallback = { this.getMovies }
          />
        </div>

        <div >
          <ShowMovieResults
            movies = { this.state.movies }
            tmbdStatusCallback = { this.statusUpdate }
          />
        </div>
      </div>
    );
  }
}

Tmbd.propTypes = {
  statusCallback: PropTypes.func.isRequired
}

export default Tmbd;
