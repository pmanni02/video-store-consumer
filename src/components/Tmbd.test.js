import React from 'react';
import { shallow } from 'enzyme';
import Tmbd from './Tmbd';

describe('Tmbd', () => {
  test('snapshot', () => {
    const wrapper = shallow(
      <Tmbd
        statusCallback={()=>{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
