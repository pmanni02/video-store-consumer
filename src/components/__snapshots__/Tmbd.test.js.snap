// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tmbd snapshot 1`] = `
<div>
  <div
    className="show"
  >
    <SearchTmbdForm
      searchCallback={[Function]}
    />
  </div>
  <div
    className="hide"
  >
    <ShowMovieResults
      movies={Array []}
      tmbdStatusCallback={[Function]}
    />
  </div>
</div>
`;
